rts/motoko-rts/src/idl.rs

```diff
@@ -488,3 +516,344 @@ unsafe extern "C" fn skip_fields(tb: *mut Buf, buf: *mut Buf, typtbl: *mut *mut
         *n -= 1;
     }
 }
+
+unsafe fn is_opt_reserved(typtbl: *mut *mut u8, end: *mut u8, t: i32) -> bool {
+    if is_primitive_type(t) {
+        return t == IDL_PRIM_reserved;
+    }
+
+    // unfold t
+    let mut t = t;
+
+    let mut tb = Buf {
+        ptr: *typtbl.add(t as usize),
+        end: end,
```

Comment:
```suggestion
        end,
```

Reply:
Hmm, I'm not sure I like that one since it confused field names with identifiers. And doesn't match the style in the rest of the file